aiohttp/client_proto.py

```diff
@@ -46,7 +46,26 @@ def __init__(self, loop: asyncio.AbstractEventLoop) -> None:
 
         self._timeout_ceil_threshold: Optional[float] = 5
 
-        self.closed: asyncio.Future[None] = self._loop.create_future()
+        self._closed: Union[None, asyncio.Future[None]] = None
+        self._connection_lost_called = False
+
+    @property
+    def closed(self) -> Union[None, asyncio.Future[None]]:
+        """Future that is set when the connection is closed.
+
+        This property returns a Future that will be completed when the connection
+        is closed. The Future is created lazily on first access to avoid creating
+        futures that will never be awaited.
+
+        Returns:
+            - A Future[None] if the connection is still open or was closed after
+              this property was accessed
+            - None if connection_lost() was already called before this property
+              was ever accessed (indicating no one is waiting for the closure)
+        """
+        if self._closed is None and not self._connection_lost_called:
+            self._closed = self._loop.create_future()
+        return self._closed
```

Comment:
Is this not a breaking change?
e.g. If `self._connection_lost_called`, then should we not create and return a resolved Future? This would obviously need us to hold on to the resolved status elsewhere though..

Or is that not a concern in this case?

Reply:
`.closed` and its not part of the public API so I don't think its a breaking change. Also we just added it in the last week by backport so I'm not so concerned about changing it.

Reply:
Its also only used in the connector